### DISCUSSION TITLE
How to proxy image with http-proxy-middleware?

### DISCUSSION THREAD

**Discussion by kopax-polyconseil** [chimurai/http-proxy-middleware#739] [General]
## Description

We are proxying a whole SPA website using Node.jS with `http-proxy-middleware` in order to edit `<meta />` tag for social network on two specific routes.

## Probleme

Using Google Chrome, this is a `.png` file retrieved without using the proxy : https://app.staging.passculture.team/images/app-icon-512px.png

This is the same `.png` file with the proxy : https://app.testing.passculture.team/images/app-icon-512px.png

We expect the MIME content type to be set like with the first URL in our proxy, instead it use `res['content-type'] = 'text/html'`

It results in Google chrome displaying this :  

```
ï¿½PNG  IHDRï¿½xï¿½ï¿½ï¿½iCCPICC profile(ï¿½}ï¿½;Hï¿½Pï¿½ï¿½ï¿½JEZï¿½ â¡:Yï¿½ï¿½8jï¿½P!ï¿½ ï¿½:ï¿½ï¿½ï¿½!4iHR\×‚ï¿½ï¿½Åªï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ ï¿½qquRtï¿½ï¿½M -bï¿½pï¿½ï¿½ï¿½ï¿½Ï¹ï¿½Bï¿½ï¿½4ï¿½cï¿½tï¿½L'b6ï¿½"ï¿½^B/"Tc2ï¿½ï¿½YIJï¿½w}ï¿½#ï¿½ï¿½ï¿½8ï¿½ï¿½ï¿½çŠ¨yï¿½ï¿½xï¿½ï¿½Mï¿½N<ï¿½iï¿½ï¿½ï¿½ï¿½ï¿½$ï¿½ï¿½ï¿½ï¿½#&
```

The proxy should behave like in the first example link. 

By reading the documentation of `http-proxy-middleware`, I don't find how to serve images without issue.

### Reproduction

```bash

git clone --depth 1 https://github.com/pass-culture/pass-culture-app-native.git -b bar
cd server
yarn
yarn dev
// then it's listening on http://localhost:8080
```

Open with chrome `http://localhost:8080/images/app-icon-512px.png`

## Question

How can I fix the proxy ?

**Comment by chimurai** [comment]
Did you try the `responseInterceptor`?

Here is an example with image manipulation:
https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/response-interceptor.md#manipulate-image-response
